insights/src/main/java/org/jboss/as/telemetry/extension/InsightsService.java

```diff
@@ -0,0 +1,490 @@
+            } catch (IOException e) {
+                ROOT_LOGGER.couldNotLoadPropertiesFile(e);
+            } finally {
+                if (fis != null) {
+                    try {
+                        fis.close();
+                    } catch (IOException e) {
+                        ROOT_LOGGER.couldNotClosePropertiesFile(e);
+                    }
+                }
+            }
+        }
+    }
+
+    private void sendJdr(String fileName, String uuid) {
+        String description = JDR_DESCRIPTION.replace("{uuid}", uuid);
+        boolean wasSuccessful = true;
+        File file = new File(fileName);
+        String systemUrl = "";
+        try {
+            systemUrl = systemEndpoint;
+            Response response = api.getInsights().get(systemUrl + uuid);
+            com.redhat.gss.redhat_support_lib.infrastructure.System system = response
+                    .readEntity(com.redhat.gss.redhat_support_lib.infrastructure.System.class);
+            // if system is unregistered then attempt to register system
+            if (system.getUnregistered_at() != null) {
+                api.getInsights().addSystem(systemUrl, uuid,
+                        InetAddress.getLocalHost().getHostName());
+            }
+        } catch (RequestException e) {
+            // if the system was not found then attempt to register the system
+            if (e.getMessage().contains(
+                    "" + Response.Status.NOT_FOUND.getStatusCode())) {
+                try {
+                    api.getInsights().addSystem(systemUrl, uuid,
+                            InetAddress.getLocalHost().getHostName());
+                } catch (RequestException exception) {
+                    ROOT_LOGGER.couldNotRegisterSystem(exception);
+                } catch (MalformedURLException exception) {
+                    ROOT_LOGGER.couldNotRegisterSystem(exception);
+                } catch (UnknownHostException exception) {
+                    ROOT_LOGGER.couldNotRegisterSystem(exception);
+                }
+            } else {
+                ROOT_LOGGER.couldNotRegisterSystem(e);
+            }
+        } catch (MalformedURLException e) {
+            ROOT_LOGGER.couldNotFindSystem(e);
+        } catch (UnknownHostException e) {
+            ROOT_LOGGER.couldNotFindSystem(e);
+        }
+        try {
+            api.getInsights().upload((insightsEndpoint + uuid), file,
+                    description);
+        } catch (FileNotFoundException e) {
+            wasSuccessful = false;
+            ROOT_LOGGER.couldNotFindJdr(e);
+        } catch (MalformedURLException e) {
+            wasSuccessful = false;
+            ROOT_LOGGER.couldNotUploadJdr(e);
+        } catch (ParseException e) {
+            wasSuccessful = false;
+            ROOT_LOGGER.couldNotUploadJdr(e);
+        } catch (RequestException e) {
+            wasSuccessful = false;
+            ROOT_LOGGER.couldNotUploadJdr(e);
```

Comment:
you could use catch mutliple given that you do same thing for different exceptions.
 } catch (RequestException|ParseException|MalformedURLException e) {